src/Roslyn.Diagnostics.Analyzers/Core/ExportedPartsShouldHaveImportingConstructorCodeFixProvider.cs

```diff
@@ -0,0 +1,189 @@
+                        break;
+
+                    case ExportedPartsShouldHaveImportingConstructor.Scenario.MultipleConstructors:
+                    default:
+                        continue;
+                }
+
+                context.RegisterCodeFix(
+                    CodeAction.Create(
+                        title,
+                        createChangedDocument,
+                        equivalenceKey: scenario),
+                    diagnostic);
+            }
+
+            return Task.CompletedTask;
+        }
+
+        private async Task<Document> AddExplicitImportingConstructorAsync(Document document, TextSpan sourceSpan, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            var exportAttribute = root.FindNode(sourceSpan, getInnermostNodeForTie: true);
+            var exportAttributeSymbol = semanticModel.GetSymbolInfo(exportAttribute, cancellationToken).Symbol?.ContainingType;
+            var importingConstructorAttributeSymbol = exportAttributeSymbol.ContainingNamespace.GetTypeMembers(nameof(ImportingConstructorAttribute)).FirstOrDefault();
+
+            var generator = SyntaxGenerator.GetGenerator(document);
+
+            var declaration = exportAttribute;
+            var declarationKind = generator.GetDeclarationKind(declaration);
+            while (declarationKind != DeclarationKind.Class)
+            {
+                declaration = generator.GetDeclaration(declaration.Parent);
+                if (declaration is null)
+                {
+                    return document;
+                }
+
+                declarationKind = generator.GetDeclarationKind(declaration);
+            }
+
+            var importingConstructor = generator.ConstructorDeclaration(
+                containingTypeName: generator.GetName(declaration),
+                parameters: Enumerable.Empty<SyntaxNode>(),
+                Accessibility.Public,
+                DeclarationModifiers.None,
+                baseConstructorArguments: null,
+                statements: Enumerable.Empty<SyntaxNode>());
+            importingConstructor = generator.AddAttributes(importingConstructor, generator.Attribute(generator.TypeExpression(importingConstructorAttributeSymbol)));
+
+            var newDeclaration = generator.AddMembers(declaration, importingConstructor);
+            return document.WithSyntaxRoot(root.ReplaceNode(declaration, newDeclaration));
+        }
+
+        private async Task<Document> MakeConstructorPublicAsync(Document document, TextSpan sourceSpan, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var importingConstructorAttribute = root.FindNode(sourceSpan, getInnermostNodeForTie: true);
```

Comment:
Bail out on null `importingConstructorAttribute`?

Reply:
Not possible unless the code fix infrastructure is broken; leaving as is since the invalid state exception is already handled by the infrastructure